app.rb

```diff
@@ -0,0 +1,165 @@
+    @main.display_menu
+  end
+
+  def quit
+    puts 'Bye!'
+    @music_albums.save_music_albums
+    @storage.save_books(@book_options.books)
+    exit
+  end
+
+  def add_game
+    puts ''
+    print 'Please enter author\'s first_name: '
+    first_name = gets.chomp
+    print 'Please enter author\'s last_name: '
+    last_name = gets.chomp
+    print 'Are multiplayers option available? (Y/N): '
+    multiplayer_game = gets.chomp.downcase == 'y'
+    print 'Please enter last_played_date (DD/MM/YYYY): '
+    last_played_date = gets.chomp
+    print 'Please enter game_published_date (DD/MM/YYYY): '
+    released_date = gets.chomp
+    game = Game.new(multiplayer_game, last_played_date, released_date)
+    author = Author.new(first_name, last_name)
+    author.add_item(game)
+    @games << { 'id' => game.id, 'multiplayer_game' => game.multiplayer, 'last_played_date' => game.last_played_at,
+                'released_date' => game.publish_date }
+    @authors << { 'first_name' => game.author.first_name, 'last_name' => game.author.last_name }
+    puts "\nSuccessfully added new game"
+    preserve_game(@games)
+    preserve_author(@authors)
+  end
+
+  def list_games
+    puts ''
+    if @games.empty?
+      puts 'No record for game found'
+    else
+      puts 'List of all Games:'
+      @games.each do |game|
+        puts '......................................................'
+        puts "Game_ID: #{game['id']}"
+        puts "Multiplayer: #{game['multiplayer_game'] ? 'Yes' : 'No'}"
+        puts "Last Played Date: #{game['last_played_date']}"
+        puts "Publish Date: #{game['released_date']}"
+        puts '......................................................'
+        puts ''
+      end
+    end
+    retrieve_data
+  end
+
+  def list_authors
+    puts ''
+    if @authors.empty?
+      puts 'No record for authors found'
+    else
+      puts 'List of all Authors:'
+      @authors.each_with_index do |author, _index|
+        puts "Author's Name: #{author['first_name']} #{author['last_name']}"
+      end
+    end
+    retrieve_data
+  end
+
+  def preserve_game(game)
+    File.write('Storage/game.json', JSON.generate(game))
+  end
+
+  def preserve_author(author)
+    File.write('Storage/author.json', JSON.generate(author))
+  end
+
+  def retrieve_data
+    retrieve_game
+    retrieve_author
+  end
+
+  def retrieve_game
+    return unless File.exist?('Storage/game.json')
+
+    @games = File.empty?('./Storage/game.json') ? [] : JSON.parse(File.read('Storage/game.json'))
+  end
+
+  def retrieve_author
+    return unless File.exist?('Storage/author.json')
+
+    @authors = File.empty?('./Storage/author.json') ? [] : JSON.parse(File.read('Storage/author.json'))
+  end
+
+  def to_json(param)
+    {
+      'id' => param.id,
+      'multiplayer' => param.multiplayer,
+      'last_played_at' => param.last_played_at,
+      'publish_date' => param.publish_date,
+      'archived' => param.archived
+    }
+  end
```

Comment:
When listing a `Game` or `music album` or `book`, it should provide all the information about that particular item ie., info about the `Author`, `Genre`, and `Label`. However, in your case, it doesn't provide all the information. Kindly fix this issue. 
Have a look at the below example when i list the books.
![image](https://github.com/Booze33/Ruby-Capstone/assets/33268927/8a59fc83-ac0b-4702-a10e-101546253d4e)
Even though there is a lot more information about the book in the storage which in itself is not complete, however, is not being displayed in the console. 
![image](https://github.com/Booze33/Ruby-Capstone/assets/33268927/98b3fa1c-d9ce-4632-9041-e6455aa58463)

Reply:
@muneebulrehman / @Metaverse-Nabeel @Booze33 

Thanks for the review. We will make the changes and aks for another review!

Reply:
Hello @muneebulrehman 

I have made the changes as per the reviews.